tidy3d/components/simulation.py

```diff
@@ -4040,14 +4055,247 @@ def subsection(
+        This function refines the mesh around lumped elements and PEC corners.
         """
         mesh_overrides = []
 
         # For now we can suggest MeshOverrideStructures for lumped elements.
         for lumped_element in self.lumped_elements:
             mesh_overrides.extend(lumped_element.to_mesh_overrides())
 
-        return mesh_overrides
+        """We suggest MeshOverrideStructures for PEC corners. Most microwave structures use planar metallic geometries with a very thin thickness. 
+        This initial fix will suggest local mesh refinement on the plane that geometries are printed and ignore the direction along the thickness.
+          """
+        
+        # Calculate the minimum number of steps per wavelength across all dimensions
+        min_steps_per_wvl = np.max([
+            self.grid_spec.grid_x.min_steps_per_wvl,
+            self.grid_spec.grid_y.min_steps_per_wvl,
+            self.grid_spec.grid_z.min_steps_per_wvl
+        ])
+
+        # Determine the mesh size based on the wavelength and configuration parameters
+        suggested_ds = self.grid_spec.wavelength/min_steps_per_wvl/fraction_to_largest_cell
+        
+        
+        # Refine mesh around PEC corners
+        for structure in self.structures:
+            if isinstance (structure.medium, PECMedium):
+                # Process and clean up the geometry
+                processed_geometry =  self.process_geometry(structure.geometry)
+                unifier = ShapeUpgrade_UnifySameDomain(processed_geometry, True, True, True)
+                unifier.Build()
+                cleaned_shape = unifier.Shape()   
+
+                # Extract unique vertices and create override boxes for each
+                vertices = self.extract_unique_vertices(cleaned_shape)
+                for vertice in vertices:
+                        override_corner_box = self.generate_override_boxes(vertice,suggested_ds,num_cells)
+                        mesh_overrides.extend(override_corner_box)
+
+        # Update grid specification to include new overrides
+        updated_grid_spec = self.grid_spec.copy(update={
+            "override_structures": list(self.grid_spec.override_structures) + mesh_overrides,
+        })
+
+
+        # Create a new simulation instance with the updated grid spec
+        sim_copy = self.copy(update={"grid_spec": updated_grid_spec})
+        return sim_copy
+
+
+    def generate_override_boxes(self, vertice, ds, cell_number):
+
+        """
+        Generates mesh override boxes both inward and outward centered at the given vertex.
+
+        Args:
+            vertex (tuple): Coordinates (x, y, z) of the vertex.
+            ds (float): Size of the step for box boundaries.
+            cell_number (int): Number of steps to extend from the vertex to create the boxes.
+
+        Returns:
+            list: A list containing MeshOverrideStructure objects for inward and outward boxes.
```

Comment:
Is there a way to accomplish this using a single MeshOverrideStructure box centered on the vertex?

Reply:
This setting is to snapping the grid line to the vertices. If we use a single MeshOverrideStructure box centered on the vertex sometimes the grid line off from the vertices.